server/accounts.go

```diff
@@ -2871,8 +2871,35 @@ func (s *Server) updateAccountClaimsWithRefresh(a *Account, ac *jwt.AccountClaim
 			alteredScope[k] = struct{}{}
 		}
 	}
+	// collect mappings that need to be removed
+	removeList := []string{}
+	for _, m := range a.mappings {
+		if _, ok := ac.Mappings[jwt.Subject(m.src)]; !ok {
+			removeList = append(removeList, m.src)
+		}
+	}
 	a.mu.Unlock()
 
+	for sub, wm := range ac.Mappings {
+		mappings := make([]*MapDest, len(wm))
+		for i, m := range wm {
+			if m.Weight == 0 {
+				m.Weight = 100
+			}
+			mappings[i] = &MapDest{
+				Subject:    string(m.Subject),
+				Weight:     m.Weight,
+				OptCluster: m.OptCluster,
+			}
+		}
+		// This will overwrite existing entries
+		a.AddWeightedMappings(string(sub), mappings...)
```

Comment:
It will replace existing entries.
AddWeightedMappings has code to replace already existing entries.
accounts.go line 636 says: // Replace an old one if it exists.